stat() is always 64 bit on musl and stat64() has been long deprecated and
eventually removed as alias for `stat()`.
--- a/subprojects/spdlog-1.8.5/include/spdlog/details/os-inl.h
+++ b/subprojects/spdlog-1.8.5/include/spdlog/details/os-inl.h
@@ -231,7 +231,7 @@
     int fd = ::fileno(f);
 #endif
 // 64 bits(but not in osx or cygwin, where fstat64 is deprecated)
-#if (defined(__linux__) || defined(__sun) || defined(_AIX)) && (defined(__LP64__) || defined(_LP64))
+#if ((defined(__linux__) && defined(__GLIBC__)) || defined(__sun) || defined(_AIX)) && (defined(__LP64__) || defined(_LP64))
     struct stat64 st;
     if (::fstat64(fd, &st) == 0)
     {
